A sample of the additional output that is present in the `twister_harness.log` file with this PR:

```
=================================== FAILURES ===================================
____________________________ test_shell_print_help _____________________________
shell = <twister_harness.helpers.shell.Shell object at 0xfa7c0b31ad50>
def test_shell_print_help(shell: Shell):
logger.info('send "help" command')
lines = shell.exec_command('help')
>       assert 1 == 0, "this breaks"
E       AssertionError: this breaks
E       assert 1 == 0
samples/subsys/testsuite/pytest/shell/pytest/test_shell.py:15: AssertionError
```

The debug logs are still included within the harness file, so this PR simply adds information to the log. The only downside is that the harness file is written twice: once by the pytest invocation, and then it is overwritten by twister after pytest exist

--- a/scripts/pylib/twister/twisterlib/harness.py
+++ b/scripts/pylib/twister/twisterlib/harness.py
@@ -401,9 +401,6 @@ class Pytest(Harness):
             '-s', '-v',
             f'--build-dir={self.running_dir}',
             f'--junit-xml={self.report_file}',
-            '--log-file-level=DEBUG',
-            '--log-file-format=%(asctime)s.%(msecs)03d:%(levelname)s:%(name)s: %(message)s',
-            f'--log-file={self.pytest_log_file_path}',
             f'--platform={self.instance.platform.name}'
         ]
         command.extend([os.path.normpath(os.path.join(
@@ -529,9 +526,9 @@ class Pytest(Harness):
         if proc.returncode in (ExitCode.INTERRUPTED, ExitCode.USAGE_ERROR, ExitCode.INTERNAL_ERROR):
             self.status = TwisterStatus.ERROR
             self.instance.reason = f'Pytest error - return code {proc.returncode}'
-            with open(self.pytest_log_file_path, 'w') as log_file:
-                log_file.write(shlex.join(cmd) + '\n\n')
-                log_file.write('\n'.join(self._output))
+        with open(self.pytest_log_file_path, 'w') as log_file:
+            log_file.write(shlex.join(cmd) + '\n\n')
+            log_file.write('\n'.join(self._output))
 
     @staticmethod
     def _update_command_with_env_dependencies(cmd):

